### PR TITLE
Add check before deserializing messages

### PR DESCRIPTION
**Pull request by zbud-msft**
When the XPUB/XSUB proxy first starts exchanging data after sockets open, the first message that is sent is called the "envelope delimiter"

https://zguide.zeromq.org/docs/chapter3/

This envelope delimeter message is of length 1 and consists of delimeter such as "#" which is what we see when we start receiving messages. We should add a check such that we ignore the initial delimeter message.

ADO#: 24851367

### CODE REVIEW

File: common/events_common.h [hunk 1]
```diff
@@ -258,7 +258,9 @@ struct serialization
     int
     deserialize(const string& s, Map& data)
```
Comment (qiluo-msft) [inline common/events_common.h:259]: >deserialize [](http://example.com/codeflow?start=4&length=11)

Could you add unit test? I think there is test gap even for non "envelope delimiter" cases.

**Review by qiluo-msft — APPROVED** [review]

**Review by qiluo-msft — APPROVED** [review]

File: tests/events_common_ut.cpp [hunk 2]
```diff
@@ -81,17 +81,20 @@ TEST(events_common, send_recv)
...
+    map<string, string> m1, m2;
 
     EXPECT_EQ(0, zmq_message_send(sock_p0, source, m));
 
     EXPECT_EQ(0, zmq_message_read(sock_p1, 0, source1, m1));
 
     EXPECT_EQ(source, source1);
     EXPECT_EQ(m, m1);
+
+    EXPECT_EQ(0, deserialize(source2, m2));
```
Comment (liuh-80) [inline tests/events_common_ut.cpp:96]: This new code in UT doesn't relate with ZMQ. the source2 not send/receive via ZMQ, why put it here?

I suggest you create a new UT:
1. send message via ZMQ
2. receiver socket add code to check and ignore "envelope delimiter"
3. In UT check the signal (the signal can be a log message) that the "envelope delimiter" been ignored and not cause error.

**Comment by renukamanavalan** [comment]
@zbud-msft - Can you please help?

**Comment by liushilongbuaa** [comment]
@zbud-msft , 
Please help check. UT is failing.
[----------] Global test environment tear-down
[==========] 6 tests from 1 test suite ran. (15649 ms total)
[  PASSED  ] 5 tests.
[  FAILED  ] 1 test, listed below:
[  FAILED  ] eventd.proxy

 1 FAILED TEST
make[2]: *** [Makefile:60: eventd-tests] Error 1
make[2]: Leaving directory '/sonic/src/sonic-eventd'
dh_auto_build: error: make -j8 "INSTALL=install --strip-program=true" returned exit code 2
make[1]: *** [debian/rules:6: build] Error 25
make[1]: Leaving directory '/sonic/src/sonic-eventd'
dpkg-buildpackage: error: debian/rules build subprocess returned exit status 2
[  FAIL LOG END  ] [ target/debs/bullseye/sonic-eventd_1.0.0-0_amd64.deb ]

**Comment by liushilongbuaa** [comment]
https://dev.azure.com/mssonic/build/_build?definitionId=1&_a=summary&branchFilter=22020

**Comment by zbud-msft** [comment]
> @zbud-msft , Please help check. UT is failing. [----------] Global test environment tear-down [==========] 6 tests from 1 test suite ran. (15649 ms total) [ PASSED ] 5 tests. [ FAILED ] 1 test, listed below: [ FAILED ] eventd.proxy
> 
> 1 FAILED TEST make[2]: *** [Makefile:60: eventd-tests] Error 1 make[2]: Leaving directory '/sonic/src/sonic-eventd' dh_auto_build: error: make -j8 "INSTALL=install --strip-program=true" returned exit code 2 make[1]: *** [debian/rules:6: build] Error 25 make[1]: Leaving directory '/sonic/src/sonic-eventd' dpkg-buildpackage: error: debian/rules build subprocess returned exit status 2 [ FAIL LOG END ] [ target/debs/bullseye/sonic-eventd_1.0.0-0_amd64.deb ]

Yes, I am checking

**Pull request merged**